rpc/src/test/scala/org/bitcoins/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,517 @@
+      assert(balance == Bitcoins(0))
+      sendTransaction.flatMap { transaction =>
+        client.getUnconfirmedBalance.map { newBalance =>
+          assert(newBalance == transaction.amount)
+        }
+      }
+    }
+  }
+
+  it should "be able to refill the keypool" in {
+    client.getWalletInfo.flatMap { info =>
+      client.keyPoolRefill(info.keypoolsize + 1).flatMap { _ =>
+        client.getWalletInfo.flatMap { newInfo =>
+          assert(newInfo.keypoolsize == info.keypoolsize + 1)
+        }
+      }
+    }
+  }
+
+  it should "be able to get the block count" in {
+    val blockCountF = client.getBlockCount
+    blockCountF.map { count =>
+      logger.info(count.toString)
+      assert(count >= 0)
+    }
+  }
+
+  it should "be able to get the connection count" in {
+    val connectionCountF = client.getConnectionCount
+    connectionCountF.map { count =>
+      logger.info(count.toString)
+      assert(count == 0)
+    }
+  }
+
+  it should "be able to get the best block hash" in {
+    val bestHashF = client.getBestBlockHash
+    bestHashF.map { hash =>
+      logger.info(hash.toString)
+      assert(true)
+    }
+  }
+
+  it should "be able to get the mining info" in {
+    val miningInfoF = client.getMiningInfo
+    miningInfoF.map { info =>
+      logger.info(info.toString)
+      assert(info.chain == "regtest")
+    }
+  }
+
+  it should "be able to get the chain tips" in {
+    val chainTipsF = client.getChainTips
+    chainTipsF.map { tipArray =>
+      tipArray.foreach { tip =>
+        logger.info(tip.toString)
+        assert(tip.status == "active")
+      }
+      assert(true)
+    }
+  }
+
+  it should "be able to get the network info" in {
+    val networkInfoF = client.getNetworkInfo
+    networkInfoF.map { info =>
+      logger.info(info.toString)
+      assert(info.networkactive)
+      assert(info.connections == 0)
+      assert(info.localrelay)
+    }
+  }
+
+  it should "be able to generate blocks" in {
+    val blocksF = client.generate(3)
+    blocksF.map { blocks =>
+      blocks.foreach(block => logger.info(block.toString))
+      assert(blocks.length == 3)
+    }
+  }
+
+  it should "be able to generate blocks and then get their serialized headers" in {
+    val blocksF = client.generate(2)
+    blocksF.flatMap { blocks =>
+      blocks.foreach(block => logger.info(block.toString))
+      val headerF1 = client.getBlockHeaderRaw(blocks(1))
+      headerF1.map { header =>
+        logger.info(header.toString)
+        assert(header.previousBlockHashBE == blocks(0))
+      }
+    }
+  }
+
+  it should "be able to generate blocks and then get their headers" in {
+    val blocksF = client.generate(2)
+    blocksF.flatMap { blocks =>
+      val headerF0 = client.getBlockHeader(blocks(0))
+      headerF0.map { header =>
+        logger.info(header.toString)
+        assert(header.nextblockhash.contains(blocks(1)))
+      }
+      val headerF1 = client.getBlockHeader(blocks(1))
```

Comment:
Note that this Future does not depend on the `headerF0` future. So this one might finish running first and then terminate the test case without testing 

```
assert(header.nextblockhash.contains(blocks(1)))
```